### PR TITLE
allow smart linking to home screen

### PR DESCRIPTION
**Pull request by Robert-Costello**
## Technical Summary

smartlink-home

This allows app builders build a smartlink with no module session endpoint that takes a user to the app home screen when they click it. The format for these links will be: 
`https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}`

This pattern will be matched in the url for the `session_endpoint` view. When view is called with an endpoint_id equaling `None` it creates a more limited `cloudcare_state` and forwards the user to the home screen of given app id. The app id is converted to current build id in the view.

jira ticket: https://dimagi-dev.atlassian.net/browse/USH-3337

## Feature Flag

SESSION_ENDPOINTS

### Safety story

Tested locally. Will also be tested on staging. All of the checks associated with `session_endpoint` view apply here.
### QA Plan

no QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by shubham1g5** [comment]
> https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}/smartlink-home

wondering if it's better to just have `https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}`. If not, maybe `https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}/home` is better.

**Comment by Robert-Costello** [comment]
>wondering if it's better to just have https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}.

using some endpoint here lets us use the `session_endpoint` view. I think otherwise there would be a lot of code repeated setting up another url and view.

>If not, maybe `https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}/home` is better.

I was using `/home` originally and I also prefer that. My only concern there was that if an app somewhere was configured with `home` as a `Session Endpoint ID` that it would cause problems. Maybe that's unlikely, though. It wouldn't make a lot of sense to have `home` as an id for a menu that isn't the home screen or for a form.

**Comment by shubham1g5** [comment]
> My only concern there was that if an app somewhere was configured with home as a Session Endpoint ID that it would cause problems

that's a fair point, We should ideally add a check on session endpoint fields to throw an error if someone uses the id we end up reserving for home.

**Comment by shubham1g5** [comment]
> that's a fair point, We should ideally add a check on session endpoint fields to throw an error if someone uses the id we end up reserving for home.

That makes me wonder if we should let user input the session endpoint id for home screen as well to be consistent with other screens. I earlier thought that we would not need to define an endpoint id at all for home screen but that doesn't seem to be true based on your comments. @orangejenny curious If you have thoughts on the design here.

**Comment by orangejenny** [comment]
I like the `https://www.commcarehq.org/a/{DOMAIN}/app/v1/{APP ID}` thought, because it's one less thing for the user to need to configure and remember. Isn't that doable with just an additional URL pattern that also calls `session_endpoint`, if you update `session_endpoint` to default the `endpoint_id` param to `None`?

**Comment by Robert-Costello** [comment]
> Isn't that doable with just an additional URL pattern that also calls `session_endpoint`, if you update `session_endpoint` to default the `endpoint_id` param to `None`?

ah, yes that sounds very doable.

File: corehq/apps/cloudcare/views.py [hunk 1]
```diff
@@ -595,13 +595,19 @@ def _fail(error):
...
-    cloudcare_state = json.dumps({
-        "appId": build_id,
-        "endpointId": endpoint_id,
-        "endpointArgs": {
-            urllib.parse.quote_plus(key): urllib.parse.quote_plus(value)
-            for key, value in request.GET.items()
-        },
-        "forceLoginAs": force_login_as,
-    })
+    if endpoint_id == "smartlink-home":
```
[the remark this answers is not in the document]
Comment (Robert-Costello) [reply corehq/apps/cloudcare/views.py:598]: I was on the fence about this and ended up going with this if/else. Happy to Dry it up, though.

[the remark this answers is not in the document]
Comment (Robert-Costello) [reply corehq/apps/cloudcare/views.py]: DRYer commit: https://github.com/dimagi/commcare-hq/pull/33283/commits/c77692d6131f74c787531331ea8077c42528a19a

**Review by orangejenny — APPROVED** [review]

**Pull request merged**